sys/xtimer/xtimer_core.c

```diff
@@ -148,27 +148,31 @@ static inline void _schedule_earliest_lltimer(uint32_t now)
 /**
  * @brief compare two timers. return true if timerA expires earlier than or equal to timerB and false otherwise.
  */
-static bool _timer_comparison(xtimer_t* timerA, xtimer_t* timerB)
+static bool _timer_comparison(xtimer_t* timerA, xtimer_t* timerB, uint32_t now)
 {
     if (timerA->long_offset < timerB->long_offset) {
         return true;
     }
-    if (timerA->long_offset == timerB->long_offset
-            /* this condition is needed for when timerA was already expired before timerB starts */
-        && (timerA->start_time + timerA->offset < timerB->start_time
-            /* it is necessary to compare two offsets, instead of two absolute times */
-            || timerA->start_time + timerA->offset - timerB->start_time <= timerB->offset)) {
-        return true;
+    if (timerA->long_offset == timerB->long_offset) {
+        uint32_t elapsedA = now - timerA->start_time;
+        uint32_t elapsedB = now - timerB->start_time;
```

Comment:
Why is `now` required for the comparison? Shouldn't `start_time` and `offset` of both timers be enough? 
Something like that:
```c
if (timerA->start_time == timerB->start_time) {
    return timerA->offset <= timerB->offset;
}
if (timerA->start_time > timerB->start_time) {
    uint32_t diff = timerA->start_time - timerB->start_time;
    if ((timerA->offset + diff) <= timerB->offset) {
        return true;
    }
    return false;
}
if (timerA->start_time < timerB->start_time) {
    uint32_t diff = timerB->start_time - timerA->start_time;
    if (timerA->offset <= (timerB->offset + diff)) {
        return true;
    }
    return false;
}
```
(or some simplified/combined version of it)

Reply:
Again, we should compare two offsets instead of two absolute times. We should avoid something like "timerA->start_time < timerB->start_time" because this does not necessarily mean that timerB starts later than timerA (e.g., timerA->start_time = 0x00000005 and timerB->start_time = 0xFFFFFFFF (5 ticks before)). "now" is an anchor point, which is later or equal to both timerA->start_time and timerB->start_time.

Reply:
You are right - I was thinking in an absolute instead of relative universe, ignoring that start_time represents only the lsb part of the the start ;)